맛집 리스트 및 한끼팟 UI 구현

mj0107 opened the pull request:
## 개요
<!---- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
- **맛집 리스트의 각 아이템 컴포넌트의 디자인이 수정됨에따라 디자인을 수정하였습니다.**
  - 각 식당의 대표메뉴를 Carousel로 볼 수 있도록 구현하였습니다.
- **오른쪽에 한끼팟 리스트를 볼 수 있는 사이드뷰를 구현하였습니다.**
  - 사이드뷰는 접어두기 버튼과 왼쪽 화살표를 누름에 따라 접고 펼칠 수 있습니다.
  - 수직 Carousel로 구현하여 아래 버튼을 누를 시 더 많은 한끼팟을 볼 수 있습니다.
- **화면 해상도에 따라 한 줄에 식당의 개수가 달라지도록 구현하였습니다.** 
  - flex-wrap을 사용하였습니다.
  - 기존 grid에서 flex로 바꾸면서, 남는 공간이 부족해 왼쪽 filter의 width가 줄어 style이 어긋나는 현상이 있었습니다.
  해당 문제는 flex-shrink를 사용해 크기를 고정하였습니다.
- **Restaurant로 시작했던 컴포넌트 명을 모두 Store로 시작하도록 바꿨습니다.**
- **Carousel 구현에 필요한 react-slick 라이브러리를 설치했습니다.**

_한끼팟과 식당 리스트가 같은 페이지에 위치함에 따라 브랜치명에 맞지않게 한끼팟도 같이 진행하였습니다..._

## PR 유형
어떤 변경 사항이 있나요?
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [x] 빌드 부분 혹은 패키지 매니저 수정
- [x] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

### 한끼팟을 펼쳤을 때
![image](https://github.com/user-attachments/assets/ce0ec8e9-2627-4914-9ea2-cbba68eb855f)

### 한끼팟을 접어두었을 때
![image](https://github.com/user-attachments/assets/5faf96da-67f8-423e-b6f6-355a58bfa297)

mj0107 committed:
design: 나의 위치 찾기 input에서 select로 변경

mj0107 committed:
chore: eslint의 spreading props 에러 무시 설정

mj0107 committed:
chore: react-slick 라이브러리 추가

mj0107 committed:
design: 예시 음식 사진 추가

mj0107 committed:
rename: Restaurant Store로 변경

mj0107 committed:
feat: 식당 대표 메뉴 이미지 컴포넌트 구현

mj0107 committed:
feat: 식당 대표 메뉴 carousel 컴포넌트 구현

mj0107 committed:
feat: 식당 대표 메뉴 carousel 화살표 컴포넌트 구현

- 화살표 방향을 props로 받아 조건부 스타일 적용

mj0107 committed:
feat: 식당 태그 컴포넌트 구현

mj0107 committed:
feat: 식당 정보 카드 컴포넌트 구현

mj0107 committed:
feat: 한끼팟 리스트 수직 carousel 컴포넌트 구현

mj0107 committed:
feat: 한끼팟 사이드뷰 구현

talentforest commented on src/components/meal-recruit/MealRecruitList.tsx:
```diff
@@ -0,0 +1,51 @@
+
+import MealRecruitListItem from '@/components/meal-recruit/MealRecruitListItem';
+
+export default function MealRecruitList() {
```
혹시 컴포넌트명 MealRecruitSlider 는 어떨까요? 리스트도 괜찮은데 슬라이드가 적용된 컴포넌트라는 걸 나타내면 더 가독성이 좋을 것 같아서요🤔

talentforest commented on src/components/meal-recruit/MealRecruitListItem.tsx:
```diff
@@ -0,0 +1,43 @@
+export default function MealRecruitListItem() {
```
저희 코드 컨벤션에서 ListItem은 빼기로 하지 않았나용...?? 여기도 Card로 써야하나...??

mj0107 replied:
아 빼기로 했던걸 하기로 한거로 헷갈렸네요!! 그럼 Slider랑 Card로 바꿀게요!

talentforest commented on src/components/meal-recruit/MealRecruitList.tsx:
```diff
@@ -0,0 +1,51 @@
+        {...settings}
+      >
+        {Array.from({ length: 10 }, (_, idx) => (
+          <div key={idx} className="py-2">
```
`div`를 빼도 되지 않을까요..?? `py-2`는 `MealRecruitListItem`의 최상단 `div`에 적용해도 되지 않을까요? 너무 세세하게 코드 리뷰하는 것 같긴한데...😅

mj0107 replied:
이게 원래 저도 처음엔 ListItem 컴포넌트의 상위 태그에 적용했었는데 스타일이 안먹더라구요... 아마 react-slick의 Slider의 기본 스타일이 지정되어있나봐요.
그래서 편법이라 볼 수도 있겠지만... div로 한번 더 감쌌습니다 ㅠㅠ

talentforest commented on src/components/meal-recruit/MealRecruitListItem.tsx:
```diff
@@ -0,0 +1,43 @@
+export default function MealRecruitListItem() {
+  return (
+    <div className="h-full w-full">
+      <div className="rounded-lg bg-white px-5 py-4 shadow-[2px_4px_12px_0px_rgba(0,0,0,0.08)]">
```
이거 그림자 쉐도우 한번 통일해서 tailwindcss.config.js에서 설정하는게 좋을 것 같기도 하네요..! 박스 그림자는 통일하는 게 좋을 것 같아서용!

mj0107 replied:
넵 따로 빼놓겠습니다~!

talentforest commented on src/components/store/StoreMenuSlider.tsx:
```diff
@@ -0,0 +1,25 @@
+import StoreMenuImage from '@/components/store/StoreMenuImage';
+import StoreMenuSliderArrow from '@/components/store/StoreMenuSliderArrow';
+
+export default function StoreMenuSlider() {
```
크 제가 자꾸 컴포넌트명으로 뭐라고 하는 것 같은데 😂 이미지만 슬라이드되는 것 같으니 StoreMenuImageSlider 어떠신지....!

mj0107 replied:
앗 좋습니다! 훨씬 명확한거 같아요!

talentforest commented on src/components/store/StoreMenuSliderArrow.tsx:
```diff
@@ -0,0 +1,26 @@
+import { BsChevronLeft, BsChevronRight } from 'react-icons/bs';
+
+export default function StoreMenuSlideArrow({
```
이 컴포넌트 Slider를 적용하는 다른 페이지에서도 공통적으로 사용할 것 같은데 `common` 폴더에 넣는 건 어떨까요? SliderArrow 이름으로 해서용!

mj0107 replied:
좋습니다! 이것두 따로 빼놓을게요!

talentforest commented on src/components/store/StoreTag.tsx:
```diff
@@ -0,0 +1,9 @@
+  text: string;
+}
+
+export default function StoreTag({ text }: StoreTagProps) {
```
이 태그도 스토어뿐만 아니라 홈이나 라운지에서도 사용되는 것 같아서 공통 컴포넌트로 옮기면 좋을 것 같습니다. 

그리고 텍스트 요소이니 div보다는 span 태그가 나을 것 같아요~!

mj0107 replied:
넵~!

talentforest reviewed (approved):
코드 확인했습니당! 일단 approve 할테니 코멘트 한번 확인해주세욥~!

mj0107 committed:
design: 공통 shadow tailwind.config에 설정

mj0107 committed:
rename: StoreMenuSliderArrow를 SliderArrow로 변경

- 공통 컴포넌트로 분리

mj0107 committed:
rename: StoreTag를 Tag로 변경

- 공통 컴포넌트로 분리

mj0107 committed:
rename: MealRecruitListItem을 MealRecruitCard로 변경

mj0107 committed:
rename: MealRecruitList를 MealRecruitSlider로 변경

mj0107 committed:
rename: StoreMenuSlider를 StoreMenuImageSlider로 변경

mj0107 committed:
rename: 바뀐 자식 컴포넌트명 반영

- MealRecruitSliderView: MealRecruitList를 MealRecruitSlider로 변경
- StoreCard: StoreTag를 Tag로 변경, StoreMenuSlider를 StoreMenuImageSlider로 변경
